### ISSUE TITLE
C-g for escape or cancel

### ISSUE THREAD

**Issue by agentgt** [linty-org/readline.nvim#6]
I'm not exactly a knowledable vim user as I'm originally more of an emacs users so take I could be wrong on this.

Anyway coming from emacs I find myself pressing C-g often and I checked readline and indeed it is in there as well.

Might be a binding worth adding?

**Comment by andmis** [comment]
I don't quite understand. What would you like `C-g` to do? Act as `<Esc>`?

You can `:map <C-g> <Esc>` and/or `:map! <C-g> <Esc>`.

**Comment by agentgt** [comment]
I guess one of the reasons I use this library over tpope's which I used to use it apparently works better with various nvim plugins. Again this is probably because I'm a vim moron.

Like I guess I'm not sure what the Lua equivalent is to mapping to escape.

**Comment by andmis** [comment]
I think global mappings to `<Esc>` are probably the best it's possible to do. The only potential gotcha I can think of is that plugins can override the mappings in specific buffers, for example. But there's nothing that this plugin can do about that.

The main reasons to use this plugin over Tim Pope's is having better control over specific mappings, Lua interop, more-faithful implementation of the Readline functions, and this plugin works in Command-line mode.

The Lua version of the mappings you want are:
```
vim.keymap.set('', '<C-g>', '<Esc>')
vim.keymap.set('!', '<C-g>', '<Esc>')
```

**Comment by agentgt** [comment]
Yes I just checked and indeed keymap.set is overloaded... Sorry I'm not used to Lua either. I should have tried that before bothering you.

**Comment by andmis** [comment]
Sounds like this is resolved, closing.
